lib/url_parser.js

```diff
@@ -10,7 +10,13 @@ module.exports = function(url, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
 
-  let result = parser.parse(url, true);
+  let result;
+  try {
+    result = parser.parse(url, true);
+  } catch (e) {
+    return callback(new Error('URL malformed, cannot be parsed'));
```

Comment:
Do we want to add any specific error on how the url is malformed?

Reply:
The original error from Node doesn't tell us that so we can't pass that info on, we'd have to be parsing the url ourselves and checking for every possible error to find out before using `url.parse`.

Reply:
Ok, makes sense